### PR TITLE
Support the new `extended tweet_mode` API

### PR DESCRIPTION
**Pull request by muesli**
Twitter's API has changed and now shortens statuses that are longer
than 140 chars, but have an acceptable length by the new Twitter
rules.

This change introduces Tweet.FullText, which contains the
unshortened form of the status. Tweet.Text's behavior should be
unchanged and will contain the shortened form. All API requests are
now executed in the new extended mode.

Sadly the streaming API now behaves slightly different to the regular
REST API and contains the extended Tweet information in a separate
struct. This change also parses this extra struct and correctly merges
the status into the Tweet struct, so both API modes behave the same
and remain backwards compatible, as far as anaconda users are concerned.

This PR makes #158 and #190 obsolete.

### CODE REVIEW

**Comment by muesli** [comment]
Will fix the detected issues.

**Comment by muesli** [comment]
There we go, I've fixed one bug and updated the test cases to match the new API responses / requests.

**Comment by muesli** [comment]
This can probably still be improved by attaching our own UnmarshalJSON method to anaconda.Tweet. Will look at that now.

**Comment by muesli** [comment]
Got a fix for retweets coming up and will squash everything then.

**Comment by muesli** [comment]
Alright, I think we're all set!

**Comment by muesli** [comment]
Looks like Travis-CI is stuck...

File: json/statuses/show.json [hunk 1]
```diff
@@ -1 +1 @@
...
..."contributors_enabled":false,"is_translator":false,"is_translation_enabled":false,"profile_background_color":"C0DEED","profile_background_image_url":"http:\/\/abs.twimg.com\/images\/themes\/theme1\/bg.png","profile_background_image_url_https":"https:\/\/abs.twimg.com\/images\/themes\/theme1\/bg.png","profile_background_tile":false,"profile_image_url":"http:\/\/pbs.twimg.com\/profile_images\/2388595262\/v02jhlxou71qagr6mwet_normal.png","profile_image_url_https":"https:\/\/pbs.twimg.com\/profile_images\/2388595262\/v02jhlxou71qagr6mwet_normal.png","profile_banner_url":"https:\/\/pbs.twimg.com\/profile_banners\/113419064\/1398369112","profile_link_color":"0084B4","profile_sidebar_border_color":"C0DEED","profile_sidebar_fill_color":"DDEEF6","profile_text_color":"333333","profile_use_background_image":true,"has_extended_profile":false,"default_profile":true,"default_profile_image":false,"following":true,"follow_request_sent":false,"notifications":false},"geo":null,"coordinates":null,"place":null,"contributors":null,"is_quote_status":false,"retweet_count":2,"favorite_count":3,"favorited":false,"retweeted":false,"possibly_sensitive":false,"possibly_sensitive_appealable":false,"lang":"en"}
```
Comment (ChimeraCoder) [inline json/statuses/show.json]: This is tough to diff visually - mind pointing me to the the difference here? Is the only difference `text`->`full_text`?

File: twitter_test.go [hunk 2]
```diff
@@ -197,7 +199,7 @@ func Test_GetTweet(t *testing.T) {
 	}
 
 	if tweet.Text != tweetText {
-		t.Fatalf("Tweet %d contained incorrect text. Received: %s", tweetId, tweetText)
+		t.Fatalf("Tweet %d contained incorrect text. Received: %s", tweetId, tweet.Text)
```
Comment (ChimeraCoder) [inline twitter_test.go]: hah - good catch!

File: json/statuses/show.json (hunk 1 above)
Comment (muesli) [reply json/statuses/show.json]: This is indeed the only difference (`s/,"text":/,"full_text":`). I'm currently writing a little script to regenerate those JSON files for us, so we can update our mocks whenever Twitter updates their API.

File: twitter_test.go (hunk 2 above)
Comment (muesli) [reply twitter_test.go]: This one gave me headaches trying to debug my changes. Until I finally noticed this mistake was misleading me the entire time ;-)

**Comment by muesli** [comment]
Rebased to upstream/master!

**Pull request merged**